### PR TITLE
actor: simplify empty ByteString instances

### PR DESCRIPTION
**Pull request by jrudolph**
A few potential improvements I found when looking at #29244.

### CODE REVIEW

File: akka-actor/src/main/scala-2.13/akka/util/ByteString.scala [hunk 1]
```diff
@@ -142,10 +142,10 @@ object ByteString {
    */
   def fromByteBuffer(buffer: ByteBuffer): ByteString = apply(buffer)
 
-  val empty: ByteString = CompactByteString(Array.empty[Byte])
+  def empty: ByteString = ByteString1C.empty
```
Comment (jrudolph) [inline akka-actor/src/main/scala-2.13/akka/util/ByteString.scala:145]: Fewer `val` means less potential for initialization issues.

File: akka-actor/src/main/scala-2.13/akka/util/ByteString.scala [hunk 2]
```diff
@@ -273,7 +275,7 @@ object ByteString {
 
   /** INTERNAL API: ByteString backed by exactly one array, with start / end markers */
   private[akka] object ByteString1 extends Companion {
-    val empty: ByteString1 = new ByteString1(Array.empty[Byte])
+    val empty: ByteString1 = new ByteString1(new Array[Byte](0), 0, 0)
```
Comment (jrudolph) [inline akka-actor/src/main/scala-2.13/akka/util/ByteString.scala:278]: Not that it matters here but `Array.empty` is quite complicated. Removed the aux constructor below that was only used here.

**Review by johanandren — COMMENTED** [review]
Looks reasonable, do we need the same improvements for the 2.12 version as well?

File: akka-actor/src/main/scala-2.13/akka/util/ByteString.scala (hunk 2 above)
Comment (johanandren) [reply akka-actor/src/main/scala-2.13/akka/util/ByteString.scala:278]: There is `Array#emptyByteArray` though, which is pretty much the same as this

Comment (jrudolph) [reply akka-actor/src/main/scala-2.13/akka/util/ByteString.scala]: Great, thanks, TIL

**Review by ennru — APPROVED** [review]
>  do we need the same improvements for the 2.12 version as well?

The error occurs only on Scala 2.13: https://github.com/akka/akka/pull/29244#issuecomment-645968792

LGTM.

**Review by johanandren — APPROVED** [review]

**Pull request merged**